releasedir/provider.go

```diff
@@ -103,6 +103,8 @@ func (p Provider) newBlobstore(dirPath string) boshblob.DigestBlobstore {
 		blobstore = boshblob.NewLocalBlobstore(p.fs, p.uuidGen, options)
 	case "s3":
 		blobstore = NewS3Blobstore(p.fs, p.uuidGen, options)
```

Comment:
The switch is necessary as they are called the same way but require different arguments,  `io.WriterAt` vs `io.Writer`, for the second argument. Hence, the two structs can't be under the same interface. As we can only call methods on structs and interfaces, we are left having to type switch based on the struct.

The difference between `io.WriterAt` and `io.Writer` (arbitrary write offset vs append-only) is significant enough to prevent adapting gcscli to accept io.WriterAt as the difference appears at the SDK level.